inst/templates/ad_adis.R

```diff
@@ -145,3 +169,136 @@ is4 <- derive_vars_merged_lookup(
   new_vars = exprs(PARAMN),
   by_vars = exprs(PARAM)
 )
+
+
+# STEP 5: PARCAT1 and CUTOFF0x derivations.
+is5 <- is4 %>%
+  mutate(
+    PARCAT1 = ISCAT,
+    # Please, define your additional cutoff values. Delete if not needed.
+    CUTOFF02 = 4,
+    CUTOFF03 = 8
+  )
+
+
+# STEP 6: AVAL, AVALU, DTYPE and SERCAT1/N derivation
+# AVAL derivation
+is5_aval <- is5 %>%
+  mutate(
+    AVAL = case_when(
+      # ISORRES values without > or <
+      DERIVED == "ORIG" & !is.na(ISSTRESN) & ISSTRESN < ISLLOQ ~ ISLLOQ / 2,
+      DERIVED == "ORIG" & !is.na(ISSTRESN) & ISSTRESN >= ISLLOQ & ISSTRESN < ULLOQ ~ ISSTRESN,
+      DERIVED == "ORIG" & !is.na(ISSTRESN) & ISSTRESN >= ULLOQ ~ ULLOQ,
+      DERIVED == "LOG10" & !is.na(ISSTRESN) & ISSTRESN < ISLLOQ ~ log10(ISLLOQ / 2),
+      DERIVED == "LOG10" & !is.na(ISSTRESN) & ISSTRESN >= ISLLOQ & ISSTRESN < ULLOQ ~ log10(ISSTRESN),
+      DERIVED == "LOG10" & !is.na(ISSTRESN) & ISSTRESN >= ULLOQ ~ log10(ULLOQ),
+      DERIVED == "4FOLD" & !is.na(ISSTRESN) & ISSTRESN < ISLLOQ ~ ISLLOQ,
+      DERIVED == "4FOLD" & !is.na(ISSTRESN) & ISSTRESN >= ISLLOQ & ISSTRESN < ULLOQ ~ ISSTRESN,
```

Comment:
@federicobaratin I found some use of `!!` in this `case_When`. I don't think this was intentional and removed from the code.

Reply:
Oh!
No, not intentional at all.
Thanks Ben!